### PR TITLE
Added the option to open the date range picker in a modal

### PR DESCRIPTION
**Pull request by FinalButterfly**
WIP

### CODE REVIEW

File: src/modules/ngx-daterange/src/components/datepicker/date-range-picker.component.ts [hunk 1]
```diff
@@ -77,7 +78,12 @@ export class DateRangePickerComponent implements OnInit {
         targetPathClassNames = event['path'].map(obj => obj.className);
       }
 
-      const containerElementClassRoot = 'dateRangePicker';
+      let containerElementClassRoot = '';
+      if(this.options.modal === true) {
```
Comment (alsoicode) [inline src/modules/ngx-daterange/src/components/datepicker/date-range-picker.component.ts]: I would change this if/else statement to a ternary

File: src/modules/ngx-daterange/src/components/datepicker/date-range-picker.component.ts [hunk 2]
```diff
@@ -196,8 +202,12 @@ export class DateRangePickerComponent implements OnInit {
   }
 
   toggleCalendarVisibility(value?: boolean): void {
-    this.showCalendars = value !== null ? value : !this.showCalendars;
-  }
+    if(this.options.modal === true) {
+      this.displayStyle = value !== null ? value === false ? 'none' : 'block' : this.showCalendars === false ? 'none' : 'block';
+    } else {
```
Comment (alsoicode) [inline src/modules/ngx-daterange/src/components/datepicker/date-range-picker.component.ts]: Please put `else` statement on newline for consistency

File: src/modules/ngx-daterange/src/components/datepicker/date-range-picker.component.html [hunk 3]
```diff
@@ -8,6 +8,85 @@
...
+      <div class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-content">
+          <div class="modal-body">
+            <div [ngClass]="{'dateRangePickerModal': true,  'dateRangePicker-single': options.singleCalendar, 'is-chrome': checkChrome(), 'mobile': isMobile }">
+              <div class="container-fluid">
+                <div class="row">
+                  <div class="col">
+        
+                    <!-- Manual Inputs -->
+                    <ng-container *ngIf="!isMobile">
```
Comment (alsoicode) [inline src/modules/ngx-daterange/src/components/datepicker/date-range-picker.component.html]: @FinalButterfly This template needs to be DRYed up. With the addition of the modal templating, there are now several sections that need so be abstracted into separate components.

Comment (alsoicode) [reply src/modules/ngx-daterange/src/components/datepicker/date-range-picker.component.html]: Also, the modal backdrop is missing when opening the datepicker as a modal.

Comment (FinalButterfly) [reply src/modules/ngx-daterange/src/components/datepicker/date-range-picker.component.html]: I thought I would keep it in the same file for simplicity, but ok, no problem. Of course, I'll add the backdrop too.

**Pull request merged**